scripts/snapshot-restore-balance/snapshot-balances.js

```diff
@@ -0,0 +1,81 @@
+const DISCARDED_ACCOUNTS = [
+        "5D9txxK9DTvgCznTjJo7q1cxAgmWa83CzHvcz8zhBtLgaLBV",
+        "5ETuZEyLnfVzQCaDM8aQCcsNnz6xjPKvQCtqynCLqwng8QLd",
+        "5D29UEzgStCBTnjKNdkurDNvd3FHePHgTkPEUvjXYvg3brJj",
+        "5DiMVAp8WmFyWAwaTwAr7sU4K3brXcgNCBDbHoBWj3M46PiP",
+        "5EhREncHsntgJaax9YQphk1xN3LxPu2Rzbz4A3g7Ut8cRXWq",
+        "5C84NU2477uHCUF1A8rHb89sP2D2ZsnquPaGa2Htv75FN9gm",
+        "5HdZjrmNAkWQhYQUPNv7YRYnT4vyQswjbNm8eXBvULNQz5wH",
+        "5HjFLKpiCStQgRm6ZM1fT1R5pLKAqQdUG3uh7pvzaQfhdFuB",
+        "5ECktCamcAtBFJirEzvvJmXFxgLMCTAejhqZwLT1Dxn2fwB1",
+        "5EZbvFqx3j7ejqBSPWseif8xL3PwoqMQHdMT8rs9qWoHcdR3"
+    ]
+
+async function takeSnapshot(api, snapshotBalancesFile) {
+    const balances = {};
+    const accounts = await api.query.system.account.keys();
+
+    for (const key of accounts) {
+        const account = key.args[0].toString();
+        if (DISCARDED_ACCOUNTS.includes(account)) continue;
+        const { data: { free: balance } } = await api.query.system.account(account);
+        balances[account] = balance.toString();
+    }
+    fs.writeFileSync(snapshotBalancesFile, JSON.stringify(balances, null, 2));
+
+    console.log('Snapshot saved.');
+}
+
+// Restore balances from the snapshot
+async function restoreBalances(api, keyring, snapshotBalancesFile, custodySeed) {
+    const balances = JSON.parse(fs.readFileSync(snapshotBalancesFile));
+    const custodyAccount = keyring.addFromUri(custodySeed);
+    let nonce = (await api.query.system.account(custodyAccount.address)).nonce.toNumber();
+
+    for (const [account, balance] of Object.entries(balances)) {
+        const transfer = api.tx.balances.transferAllowDeath(account, balance);
+        try {
+            const hash = await transfer.signAndSend(custodyAccount, { nonce });
+            console.log(`Transferred to ${account}, transaction hash: ${hash.toHex()}`);
+            nonce++;
+            // Add a small delay between transactions to avoid priority issues
+            await new Promise(resolve => setTimeout(resolve, 500));
+        } catch (error) {
+            console.error(`Failed to transfer to ${account}:`, error);
+        }
+    }
+}
+
+async function main() {
+    const provider = new WsProvider(WS_ENDPOINT);
+    const api = await ApiPromise.create({ provider });
+    const keyring = new Keyring({ type: 'sr25519' });
+
+    const [,, command, arg1, arg2] = process.argv;
+
+    if (command === 'snapshot') {
+        if (!arg1) {
+            console.error('Please provide the snapshot file name.');
+            process.exit(1);
+        }
+        await takeSnapshot(api, arg1);
+    } else if (command === 'restore') {
+        if (!arg1 || !arg2) {
+            console.error('Please provide the custody seed and snapshot balances file name.');
+            process.exit(1);
+        }
+        await restoreBalances(api, keyring, arg2, arg1);
+    } else {
+        console.error('Unknown command. Use "snapshot" or "restore".');
```

Comment:
It would be very helpful to have a very brief usage example, to clarify what parameters should be passed without having to look at the code.

Reply:
Adding examples in the error message